src/ui/states/state-needs-payment-info.svelte

```diff
@@ -67,8 +67,13 @@
     });
 
     if (result.error) {
-      // payment failed, notify user
       processing = false;
+
+      if (!result.error.setup_intent?.last_setup_error) {
```

Comment:
This ensures that the `setup_intent.setup_intent_failed` is still not sent, so it's safe to reuse the same Stripe Elements instance

Reply:
Hmm is this the recommended way to detect closure of the Google/Apple pay prompts? It would be great to have something more explicit, but not familiar with the Stripe API... Would it be possible to use [status](https://docs.stripe.com/api/setup_intents/object#setup_intent_object-status)?

Reply:
Also should we assume that `setup_intent will be there? As in maybe we can do:
```suggestion
      if (result.error.setup_intent !== undefined && !result.error.setup_intent?.last_setup_error) {
```
or something like this?